### PR TITLE
implements preferred and avoided paths.

### PR DESCRIPTION
**Pull request by ivukotic**
it also changes helm chart. fixes documentation.

### CODE REVIEW

File: transformer_sidecar/src/transformer_sidecar/transformer.py [hunk 1]
```diff
@@ -154,6 +154,43 @@ def prepend_xcache(file_paths):
         prefixed_paths.append(f'root://{prefix_list[pinned_xcache_index]}//{f}')
     return prefixed_paths
 
+
+def custom_path_sorting(file_paths):
```
Comment (BenGalewsky) [inline transformer_sidecar/src/transformer_sidecar/transformer.py:158]: This complicated function needs doc string to explain what it's doing.

File: transformer_sidecar/src/transformer_sidecar/transformer.py [hunk 2]
```diff
@@ -154,6 +154,43 @@ def prepend_xcache(file_paths):
...
+
+    if preferred:
+        preferred_list = [p.strip() for p in preferred.split(',')]
+
+    if avoided:
+        avoided_list = [p.strip() for p in avoided.split(',')]
+
+    for f in file_paths:
+        skip = False
+        for av in avoided_list:
```
Comment (BenGalewsky) [inline transformer_sidecar/src/transformer_sidecar/transformer.py:179]: The avoided list could be implemented more cleanly with the python `filter` function:
```python
acceptable_paths = list(filter(lambda x: x not in avoided_list, file_paths))
```

File: transformer_sidecar/src/transformer_sidecar/transformer.py [hunk 3]
```diff
@@ -154,6 +154,43 @@ def prepend_xcache(file_paths):
...
+            if av in f:
+                skip = True
+        if skip:
+            continue
+        pref = False
+        for pr in preferred_list:
+            if pr in f:
+                sorted_paths.append(f)
```
Comment (BenGalewsky) [inline transformer_sidecar/src/transformer_sidecar/transformer.py:187]: Preferred paths could also be done with a filter

File: transformer_sidecar/src/transformer_sidecar/transformer.py [hunk 4]
```diff
@@ -154,6 +154,43 @@ def prepend_xcache(file_paths):
...
+                pref = True
+        if not pref:
+            unsorted_paths.append(f)
```
Comment (BenGalewsky) [inline transformer_sidecar/src/transformer_sidecar/transformer.py:190]: unsorted paths could be implemented with a filter

**Review by BenGalewsky — CHANGES_REQUESTED** [review]
The `custom_path_sorting` needs 100% unit test coverage. It's easy enough to write tests on this standalone function to verify that it works as expected in all cases.

The code could be much easier to understand with the python filter functions

The function needs a complete docstring to explain what is going on

File: transformer_sidecar/src/transformer_sidecar/transformer.py (hunk 2 above)
Comment (ponyisi) [reply transformer_sidecar/src/transformer_sidecar/transformer.py:179]: Or list comprehension ;-) :

```
acceptable_paths = [x for x in file_paths if not any(a in x for a in avoided_list)]
```

Comment (ivukotic) [reply transformer_sidecar/src/transformer_sidecar/transformer.py:179]: I am sorry for offending your coding styles. I will delete this PR.

**Pull request closed without merging**